packages/ui/tsup.config.ts

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable import/no-nodejs-modules */
+import { execSync } from 'node:child_process';
+import { createHash } from 'node:crypto';
+import { sassPlugin, postcssModules } from 'esbuild-sass-plugin';
+import { defineConfig } from 'tsup';
+
+let gitHeadSha;
+try {
+	gitHeadSha = execSync( 'git rev-parse HEAD' ).toString();
+} catch {}
+
+export default defineConfig( {
+	entry: [ 'src/index.ts' ],
+	clean: true,
+	splitting: true,
+	experimentalDts: true,
+	sourcemap: true,
+	format: [ 'esm', 'cjs' ],
+	outDir: 'dist',
+	esbuildPlugins: [
+		sassPlugin( {
+			filter: /\.module\.(css|scss)$/,
+			embedded: true,
+			transform: postcssModules( {
+				generateScopedName: ( name, filename ) => {
+					const hash = createHash( 'md5' )
+						.update( name + filename + gitHeadSha )
+						.digest( 'hex' )
+						.slice( 0, 6 );
+					return `a8cui-${ hash }`;
```

Comment:
No strong opinion on whether we hash using the CSS content or the git hash. I think it would be generally safer to change classes for every release, since most setups won't be utilizing the class stability even if we keep some unchanged styles names unchanged across releases. (For example, I doubt a single app bundle is going to load multiple versions of the `@automattic/ui` package and then dedupe the stable styles.)

Reply:
I'm taking inspiration from @aduth's [take](https://github.com/Automattic/wp-calypso/pull/103606/files#diff-0ae4d8e9a04e920534d0906094fd70ec7111968884b76a83eb901f9830877831) for the git hash, but increasing the length a bit since we're not prefixing the class with the local name. Apparently six characters of hex (24 bits) is safe for up to ~50,000 unique class names per build.

Reply:
Since we're hashing, we may as well go all the way and use CSS content?

Reply:
I'd prefer to avoid the `git` commit hash: builds should have identical output if the source is identical. Here we have exactly the opposite, where every rebase and every new commit to the repo, no matter how unrelated, will change the classnames.

Reply:
I don't feel too strongly about this, but am somewhat curious to understand the specific reasons that the private APIs in Gutenberg were considered a failure. From the outside, I'd imagine it would be a failure because it's too easy to opt-in and successfully use features without any consequence or maintenance, since the opt-in string is relatively stable. In that sense, it could have benefitted from the constant chaos of something like what we're considering here, build reproducibility notwithstanding. This was the same reason I tried to push for [guaranteed, consistent breakage in Gutenberg deprecations](https://github.com/WordPress/gutenberg/pull/5491) early on.

Reply:
My concern here was not about obfuscation toward consumers, but whether it would cause issues when different plugins load multiple versions of the package in a single view. It's probably fine either way though.